pkg/reconciler/pipelinerun/metrics.go

```diff
@@ -56,13 +57,18 @@ type Recorder struct {
 	pipelineRun tag.Key
 	namespace   tag.Key
 	status      tag.Key
+
+	ReportingPeriod time.Duration
 }
 
 // NewRecorder creates a new metrics recorder instance
 // to log the PipelineRun related metrics
 func NewRecorder() (*Recorder, error) {
 	r := &Recorder{
 		initialized: true,
+
+		// Default to 30s intervals.
+		ReportingPeriod: 30 * time.Second,
```

Comment:
It could be nice if we keep this configurable? 🤔

Reply:
It's a public field, so it can be changed after the resource is created, but I see that this is just consumed by NewController without really having a place to hook into.   I'm happy to whatever is conventional for Tekton here.

Reply:
I'm not sure as well(I was away for some time). It's better if @afrittoli @vdemeester comment more on it.
In my opinion, we can have it as it is. If required in the future we can provide the relevant configuration option :)

However, maybe it's nice to document it? so users are aware of 30 seconds window to get updated metrics